### PR TITLE
Changing the handled operators in bv2int preprocessing pass

### PR DESCRIPTION
**Pull request by yoni206**
Some of the bit-vector operators are directly translated to integers, while others are eliminated before the translation.
This PR changes the set of operators that we eliminate (and as a consequence, also the set of operators that we handle directly):
1. The only bit-wise operator that is translated is `bvand`. The rest are now eliminated.
2. `bvneg` is now eliminated.
3. The various `division` operators are still eliminated, but using different rewrite rules.
4. zero-extend and sign-extend are now handled directly.
5. shifting is changed to favor ITEs over non-linear multiplication.

### CODE REVIEW

File: src/preprocessing/passes/bv_to_int.cpp [hunk 1]
```diff
@@ -190,24 +180,27 @@ Node BVToInt::eliminationPass(Node n)
...
                                   RewriteRule<SltEliminate>,
                                   RewriteRule<SgtEliminate>,
-                                  RewriteRule<SgeEliminate>,
-                                  RewriteRule<ShlByConst>,
-                                  RewriteRule<LshrByConst> >::apply(current);
+                                  RewriteRule<SgeEliminate>>::apply(current);
       // save in the cache
       d_eliminationCache[current] = currentEliminated;
-      // also assign the eliminated now to itself to avoid revisiting.
+      // also assign the eliminated now to itself to avoid revisiting.
```
Comment (ajreynol) [inline src/preprocessing/passes/bv_to_int.cpp:203]: Is there a format issue here with new space?

File: test/regress/regress2/bv_to_int_shifts.smt2 [hunk 2]
```diff
@@ -1,17 +1,18 @@
 ; COMMAND-LINE: --solve-bv-as-int=sum --bvand-integer-granularity=1 --no-check-models  --no-check-unsat-cores
 ; EXPECT: sat
 (set-logic QF_BV)
-(declare-fun s () (_ BitVec 64))
-(declare-fun t () (_ BitVec 64))
-(declare-fun splust () (_ BitVec 64))
-(declare-fun shift1 () (_ BitVec 64))
-(declare-fun shift2 () (_ BitVec 64))
-(declare-fun negshift1 () (_ BitVec 64))
+(declare-fun s () (_ BitVec 4))
```
Comment (ajreynol) [inline test/regress/regress2/bv_to_int_shifts.smt2:4]: Why this change?

**Review by ajreynol — CHANGES_REQUESTED** [review]
Mostly looks good, have a few minor questions.

File: src/preprocessing/passes/bv_to_int.cpp (hunk 1 above)
Comment (yoni206) [reply src/preprocessing/passes/bv_to_int.cpp]: fixed [fc48ecc55625bb34937d7ec1eab62f8732857d7b](https://github.com/CVC4/CVC4/pull/4970/commits/fc48ecc55625bb34937d7ec1eab62f8732857d7b)

File: test/regress/regress2/bv_to_int_shifts.smt2 (hunk 2 above)
Comment (yoni206) [reply test/regress/regress2/bv_to_int_shifts.smt2:4]: just to make the test run faster. 4 bits is interesting enough for these operators in terms of correctness / debug assertions.

**Review by ajreynol — APPROVED** [review]
